### ISSUE TITLE
SQL Deadlock

### ISSUE THREAD

**Issue by stroomdev10** [gchq/stroom#4408]
Looks like Deadlock when processor_task cleanup was running
[20240827.txt](https://github.com/user-attachments/files/16759532/20240827.txt)

**Comment by at055612** [comment]
```sql
TXN1
update `processor_task`
set `processor_task`.`fk_processor_node_id` = 1,
`processor_task`.`status` = 1,
`processor_task`.`status_time_ms` = 1724739464858,
`processor_task`.`start_time_ms` = null,
`processor_task`.`end_time_ms` = null,
`processor_task`.`version` = (`processor_task`.`version` + 1)
where (`processor_task`.`id` in ( .......... )

TXN2
delete from `processor_task`
where ((`processor_task`.`status` = 10 or `processor_task`.`status` = 99)
and (`processor_task`.`status_time_ms` is null or `processor_task`.`status_time_ms` < 1724134695555))
```

**Comment by at055612** [comment]
Fixed in >7.4.13
